### ISSUE TITLE
Search box does not find correspondences in various entities and ontologies

### ISSUE THREAD

**Issue by onoff0** [openlumify/openlumify#13]
Hì, by running openlumify and trying the various functions I realized that the SEARCH box does not find correspondences in the various entities and ontologies. Another problem is the automatic time setting which must be set manually when starting openlumify.
Is there any chance to solve these problems? 
Thank you !

**Comment by glennji** [comment]
Hey `onoff0`, nice to meet you! For full situational awareness, it's early days in my attempts to get this project back on track, but I can absolutely try to fix any and all bugs you find. Did you use lumify back in the good ol' days, by the way?

Let's use this issue report for the search box and I'll create another one for automatic time setting. (Will ask you some questions there.)

For search box: what exactly are you searching for, and what results are you expecting (but not receiving)?

**Comment by glennji** [comment]
Also: mind if i add you to the project? Assistance with finding bugs like this would be super helpful.

**Comment by onoff0** [comment]
Yes, the pleasure is all mine! and sorry for my english! I will try to make myself better understood.
yes I've used both lumify and visallo in the past!
Regarding the search box, in the latest versions of the open source branch of this project it 
already had bugs, unlike the first versions that worked correctly.
basically, the search box should find everything that present in the database or in the workspace, typically adding operators  
"all" `*` before or after word to search for.
(You'll notice only find matches by filtering the entities leaving the operator `*` in the default box)

example: I want to search for the title of all ontologies that will start with **lon*** or end with ***don** 
and I will have all matching words like "london" as results.




![seachbox](https://user-images.githubusercontent.com/22854543/72680445-3c680700-3aba-11ea-9cec-692df5252deb.png)

**Comment by glennji** [comment]
Okay, cool, I can see that now. Will try to debug it this week. (For the avoidance of doubt, I've got nothing to do with any of the Visallo team, nor Lumify in any previous incarnation -- I just want to revive the project and keep it opensource!)

In the logs I can see:
```
2020-01-19 22:56:17.768/AEDT DEBUG [USER_b8b0a8502da04a34b7a0258a2b3f924e:visallo] [model.search.VertexiumObjectSearchRunnerWithRelatedBase] search *Black (relatedToVertexIds: , elementExtendedData: null)
[]
2020-01-19 22:56:17.884/AEDT DEBUG [USER_b8b0a8502da04a34b7a0258a2b3f924e:visallo] [vertexium.elasticsearch5.ElasticsearchSearchQueryBase] elasticsearch results 0 of 0 (time: 10ms)
```

... and it looks like the query is being passed pretty verbatim into Elasticsearch. I'll keep digging.

**Comment by onoff0** [comment]
(I also find the project interesting and that it must continue open source!)

**Comment by onoff0** [comment]
I also tried to integrate the old versions in which search box worked correctly, but I didn't get satisfactory results!  
Is there any news?

**Comment by glennji** [comment]
Okay, so I left this one alone for the last month but have been doing a bit of experimentation with Elasticsearch and can't actually see how this _ever_ worked. :-) 

Over the next week or so I'll try to put some time into a proper fix.

**Comment by onoff0** [comment]
excellent I will also try in the week.
If it can be useful
on some versions the search text only works with entries after importing the .csv files!

**Comment by onoff0** [comment]
the other day i did some testing, and i noticed that after upgrading to **elasticsearch5** it creates the known search problems. 
I believe (but it is only a hypothesis) that the problem could be in 

**config/ visallo-elasticsearch-inprocess-properties**


**after update**
```
#
# Vertexium
#
graph.search=org.vertexium.elasticsearch5.Elasticsearch5SearchIndex
graph.search.inProcessNode=true
graph.search.indexName=visallo
graph.search.inProcessNode.homePath=${VISALLO_DIR}/datastore/elasticsearch

```
**before update**
```
#
# Vertexium
#
graph.search=org.vertexium.elasticsearch.ElasticsearchSingleDocumentSearchIndex
graph.search.inProcessNode=true
graph.search.indexName=vertexium
graph.search.inProcessNode.dataPath=${VISALLO_DIR}/datastore/elasticsearch/data/
graph.search.inProcessNode.logsPath=${VISALLO_DIR}/datastore/elasticsearch/logs/
graph.search.inProcessNode.workPath=${VISALLO_DIR}/datastore/elasticsearch/work/

```
however, by making a breakpoint during the search, it would seem that **jquery.js** **withSearch.js** and **search.js** skip the cycle returning to the null value.
I  attach a comparison of the query of a working and a non-working version.


![annotated Query](https://user-images.githubusercontent.com/22854543/77067258-f6112480-69e4-11ea-9d6f-dd105b50ce2f.jpg)

It would be interesting to fix the bug, also because it can be very useful :)
